site/en/blog/show-picker/index.md

```diff
@@ -0,0 +1,203 @@
+authors:
+  - beaufortfrancois
+date: 2022-01-11
+hero: image/vvhSqZboQoZZN9wBvoXq72wzGAf1/zWT5kk3Q5RlDNqGkdgis.jpeg
+alt: A photo of a monthly schedule printed on white paper
+tags:
+  - chrome-99
+---
+
+For a long time, you had to resort to custom widget libraries or hacks to show a
+date picker. The web platform now ships with `showPicker()`, a canonical way to
+show a browser picker not only for date, but also time, color, and files.
+
+## Background {: #background }
+
+A [very common request] we hear from web developers is:
+
+<blockquote>
+  <p>
+    How do I programmatically<br/>
+    show a picker for controls like date?
+  </p>
+  <cite>
+    Stack Overflow
+  </cite>
+</blockquote>
+
+Current answers are not great: they rely on external libraries, CSS hacks, or
+specific browser behaviors like simulating a user interaction with `click()`.
+
+I'm happy to share that those days will be over soon as the web platform is
+introducing a canonical way to show a browser picker for `<input>` elements with
+these types: `"date"`, `"month"`, `"week"`, `"time"`, `"datetime-local"`,
+`"color"`, and `"file"`. It is also working for `<input>` elements with
+suggestions powered by `<datalist>` or `"autocomplete"` which we'll cover as
+well in this article.
+
+<figure class="w-figure">
+{% Img src="image/vvhSqZboQoZZN9wBvoXq72wzGAf1/uh0U2YQnUMato21MzzbR.png", alt="Screenshots of browser pickers", width="800", height="217" %}
+  <figcaption class="w-figcaption">Browser date pickers in Chrome Desktop, Chrome Mobile, Safari Desktop, Safari Mobile, and Firefox Desktop (July 2021).</figcaption>
+</figure>
+
+{% Aside %}
+Unlike a third-party picker widget, a browser picker is familiar to the user,
+has great specific platform support, and is always maintained as part of the
+browser.
+{% endAside %}
+
+## How to show a picker {: #how-to }
+
+Calling `showPicker()` on an `<input>` element returns a promise that resolves
+when a browser picker is shown to the user. It must be called in response to a
+user gesture such as a touch gesture or mouse click; otherwise it will reject.
+For security reasons,it will also reject when it's called in a cross-origin
+iframe.
+
+A browser picker is shown when the `<input>` element is one of these types:
+`"date"`, `"month"`, `"week"`, `"time"`, `"datetime-local"`, `"color"`, and
+`"file"`.
+
+The example below shows you how to open a browser date picker.
+
+```html/9
+<input type="date">
+<button>Show the date picker</button>
+
+<script>
+  const button = document.querySelector("button");
+  const dateInput = document.querySelector("input");
+
+  button.addEventListener("click", async () => {
+    try {
+      await dateInput.showPicker();
+      // A date picker is shown.
+    } catch (error) {
+      // Use external library when this fails.
+    }
+  });
+</script>
+```
+
+A browser picker may also be shown if the `<input>` element has some suggestions
+powered by `<datalist>` or `"autocomplete"`.
```

Comment:
Are you trying to tell me that the picker is populated with data provided by `<datalist>` or `"autocomplete"`? 

Maybe something like: "A browser picker can be prepoulated with items from `<datalist>` or `"autocomplete"`."

Reply:
Yes. Done